kickstarter/android-oss
# NT-1654:Detect user validated flow

# 📲 What
- Analice the response from the login endpoint to check if a user is validatedEmail or not.
- In case feature flag on the login flow / create account flow will not continue, next steps will be presenting Interstitial screen.
- In case feature flag off the login flow / create account flow will continue as it is now.

# 🛠 How
- Added field `isEmailVerified` to the User dataModel
- Moved some previously existing UI extensions to a package `ui.extensions`
- Added logic on LoginViewModel
- Added logic on SignUpViewModel

# 👀 See
- No UI updates in this PR, bet here a screenshoot from a successful networking response, with isEmailVerified field true :) 
<img width="1125" alt="Screen Shot 2020-11-03 at 10 49 47 AM" src="https://user-images.githubusercontent.com/4083656/98315463-1817e100-1f8d-11eb-8cd6-65d62ef3af7e.png">
|  |  |

# 📋 QA

UAT Login flow: 
Try login flow from DiscoveryActivity, UpdatesActivity, PledgeFragment
- Log in with feature flag on and user with email validated -> behaviour as expected now
- Log in with feature flag on and user without email validated -> You cannot go over the logIn screen (in the future you will be presented with the Interstitial Screen)
- Log in with feature flag on and user without email validated -> try to log in, kill the app, put the app in background or kill it, open it again, you should not be logged in 
- Log in with feature flag off and user without email validated -> try to log in, kill the app, put the app in background or kill it, open it again, you should be logged in with the invalidated user
- Log in with feature flag off and user with email validated -> behaviour as expected now

UAT Create Account flow: 
- Create Account in with feature flag off  -> behaviour as expected now
- Create Account in with feature flag on  ->  You cannot go over the Creation Account Screen (in the future you will be presented with the Interstitial Screen)

# Story 📖

[NT-1638](https://kickstarter.atlassian.net/browse/NT-1638)
